roseau/load_flow/plotting.py

```diff
@@ -0,0 +1,289 @@
+"""Plotting functions for `roseau.load_flow`."""
+
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+
+import roseau.load_flow as rlf
+from roseau.load_flow.typing import Complex, ComplexArray, Float
+
+if TYPE_CHECKING:
+    import folium
+    from matplotlib.axes import Axes
+
+    FeatureMap = dict[str, Any]
+    StyleDict = dict[str, Any]
+
+_COLORS = {"a": "#234e83", "b": "#cad40e", "c": "#55b2aa", "n": "#000000"}
+_COLORS.update(
+    {
+        "an": _COLORS["a"],
+        "bn": _COLORS["b"],
+        "cn": _COLORS["c"],
+        "ab": _COLORS["a"],
+        "bc": _COLORS["b"],
+        "ca": _COLORS["c"],
+        "zero": _COLORS["a"],
+        "pos": _COLORS["b"],
+        "neg": _COLORS["c"],
+    }
+)
+
+
+#
+# Utility functions
+#
+def _get_rot(vector: Complex) -> float:
+    """Get the rotation of a vector in degrees."""
+    rot = np.angle(vector, deg=True)
+    if rot > 90:
+        rot -= 180
+    elif rot < -90:
+        rot += 180
+    return rot  # type: ignore
+
+
+def _get_align(rot: Float) -> tuple[str, str]:
+    """Get the horizontal and vertical alignment corresponding to a rotation angle."""
+    if 45 < abs(rot) < 135:
+        return "right", "center"
+    else:
+        return "center", "bottom"
+
+
+def _configure_axes(ax: "Axes", vector: ComplexArray) -> None:
+    """Configure the axes for a plot of complex data."""
+    center = np.mean(vector)
+    ax_lim = max(abs(vector - center)) * 1.2
+    ax.grid()
+    ax.set_axisbelow(True)
+    ax.set_aspect("equal")
+    ax.set_xlim(-ax_lim + center.real, ax_lim + center.real)
+    ax.set_ylim(-ax_lim + center.imag, ax_lim + center.imag)
+
+
+def _draw_voltage_phasor(ax: "Axes", potential1: Complex, potential2: Complex, color: str) -> None:
+    """Draw a voltage phasor between two potentials."""
+    voltage = potential1 - potential2
+    midpoint = (potential1 + potential2) / 2
+    ax.arrow(potential2.real, potential2.imag, voltage.real, voltage.imag, color=color)  # type: ignore
+    rot = _get_rot(voltage)
+    ha, va = _get_align(rot)
+    ax.annotate(f"{abs(voltage):.0f}V", (midpoint.real, midpoint.imag), ha=ha, va=va, rotation=rot)  # type: ignore
+
+
+#
+# Phasor plotting functions
+#
+def plot_voltage_phasors(
```

Comment:
In the future I am gonna factorize this and add `plot_current_phasors` function as well but this is a good place to start for now.